### PR TITLE
Add test for filters with spaces

### LINKED ISSUE #210: Test `filter` in search method with a value containing spaces
Add a test to check the `filter` parameter can work with values containing spaces

Example:
```
"filter": "genre = 'sci fi'"
```

Related to https://github.com/meilisearch/integration-guides/issues/30

### PR DESCRIPTION
**Pull request by Ethenyl**
# Summary

Added a unit test for search queries containing a filter with a value containing a space.

## Miscellaneous

Fixes #210.

### CODE REVIEW

**Comment by curquiza** [comment]
@Ethenyl, we are looking at this CI problem, thanks for your PR :)

**Comment by curquiza** [comment]
Thank you so much @Ethenyl!
If you do not succeed in running Docker on your machine, you can
- [download MeiliSearch with curl](https://docs.meilisearch.com/learn/getting_started/installation.html)
```bash
# Install MeiliSearch
curl -L https://install.meilisearch.com | sh
```
- launch MeiliSearch
```bash
./meilisearch --master-key=masterKey
```

- [run the tests](https://github.com/meilisearch/meilisearch-swift/blob/main/CONTRIBUTING.md#tests-and-linter-)
```bash
swift test
```

**Comment by Ethenyl** [comment]
@curquiza sorry for bothering but it seems that something is not working correctly with `filter`.

I've fixed the integration test for `Search` with a `filter` on an empty space value (e.g. `genres = 'High fantasy'`) which should return at least one item.

It doesn't, so I've tried using only `High` or `fantasy`, it didn't fetch anything. But using a `filter` on the `id` worked and sent me the right book.

It seems that for fields with a single value (like `id`) it's working well, but for others like `genres` (which can contain multiple values) it is failing.

Is it working for languages other than Swift?

**Comment by curquiza** [comment]
bors try

File: Tests/MeiliSearchIntegrationTests/SearchTests.swift [hunk 1]
```diff
@@ -666,10 +666,44 @@ class SearchTests: XCTestCase {
...
+
+  func testSearchFilterWithEmptySpace() {
+    let expectation = XCTestExpectation(description: "Search for Books using filters with a space in the value")
+
+    configureFilters {
+      typealias MeiliResult = Result<SearchResult<Book>, Error>
 
+      let query = ""
+      let limit = 5
+      let filter = "genres = 'High fantasy'"
```
[the remark this answers is not in the document]
Comment (bidoubiwa) [reply Tests/MeiliSearchIntegrationTests/SearchTests.swift:681]: The error came from an invisible char in the document. Which is very unfortunate as it was the only one :') 

Screenshot 2021-11-11 at 11 05 37

**Comment by bidoubiwa** [comment]
Thanks so much, @Ethenyl for adding these tests and for your contribution to MeiliSearch overall. 

The failing test was originated from an invisible character. Unfortunately, to remove it I need to change a part of the code that is not reviewable in this PR. 
I'm merging your PR in another PR before merging it to main so I can fix it.

**Comment by Ethenyl** [comment]
@bidoubiwa no problems, the goal is to make the software better in the end 😉

**Pull request merged**